src/AutoRest.CSharp/Mgmt/Generation/ResourceContainerWriter.cs

```diff
@@ -279,14 +279,28 @@ private void WriteGetVariants(RestClientMethod method)
 
         private void WriteListVariants()
         {
-            if (_resourceContainer.ListMethod != null)
+            if (_resourceContainer.ListMethods != null)
             {
-                WriteList(_writer, false, _resource.Type, _resourceContainer.ListMethod, $".Select(value => new {_resource.Type.Name}({ContextProperty}, value))");
-                WriteList(_writer, true, _resource.Type, _resourceContainer.ListMethod, $".Select(value => new {_resource.Type.Name}({ContextProperty}, value))");
-            }
+                foreach (var listMethod in _resourceContainer.ListMethods)
+                {
+                    if (listMethod.PagingMethod != null)
+                    {
+                        WriteList(_writer, false, _resource.Type, listMethod.PagingMethod, "List", $".Select(value => new {_resource.Type.Name}({ContextProperty}, value))");
+                        WriteList(_writer, true, _resource.Type, listMethod.PagingMethod, "List", $".Select(value => new {_resource.Type.Name}({ContextProperty}, value))");
+                    }
+
+                    if (listMethod.ClientMethod != null)
+                    {
+                        _writer.Line();
+                        WriteClientMethod(_writer, listMethod.ClientMethod, _resourceContainer.GetDiagnostic(listMethod.ClientMethod.RestClientMethod), _resourceContainer.OperationGroup, _context, true);
+                        WriteClientMethod(_writer, listMethod.ClientMethod, _resourceContainer.GetDiagnostic(listMethod.ClientMethod.RestClientMethod), _resourceContainer.OperationGroup, _context, false);
+                    }
+                }
 
-            WriteListAsGenericResource(async: false);
-            WriteListAsGenericResource(async: true);
+                // WriteRemainingPagingMethods();
+                WriteListAsGenericResource(async: false);
+                WriteListAsGenericResource(async: true);
```

Comment:
I think these still exist even if the ListMethods is empty, this is a separate API that we are exposing on each resource type for convenience, but allows you to generically request resources by resourceType but doesn't rely on the current resources RestClient at all.